### DISCUSSION TITLE
V3 Class "Filament\PanelProvider" not found

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#7213] [Help]
I get the following error during the update process:

In AdminPanelProvider.php line 22:

  Class "Filament\PanelProvider" not found

Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

**Comment by mondherid** [comment]
add  AdminPanelProvider to list of providers in config/app.php ('providers'=[......... ,' AdminPanelProvider::class'])
it is not montioned in the documentation

**Reply by jcgmalave** [reply]
AdminPanelProvider::class, is already in the providers of config/app.php as App\Providers\Filament\AdminPanelProvider::class, the error says: In AdminPanelProvider.php line 22:

  Class "Filament\PanelProvider" not found.

the line 22 of AdminPanelProvider is the class declaration which extends from PanelProvider.

Thanks in advanced.

**Reply by leandrocfe** [reply]
Is there vendor/filament/filament/src/PanelProvider.php class in your project?
what is the filament version when you run `php artisan about`?
run `composer update` again

**Reply by jcgmalave** [reply]
this file  vendor/filament/filament/src/PanelProvider.php is not in my repo, i run composer update again and it shows the same error. 

Previously i run this command  vendor/filament/filament/src/PanelProvider.php

**Reply by leandrocfe** [reply]
comment the `App\Providers\Filament\AdminPanelProvider::class` in your config/app.php and run `php artisan about`
let me know what is the Filament version..

**Reply by jcgmalave** [reply]
i comment it, and i run the php artisan about and i get the following:

In ProviderRepository.php line 206:
                                            
  Class "Filament\PanelProvider" not found

**Reply by jcgmalave** [reply]
i did it, i get the following: 

  Filament ............................................................................................................................  
  Packages ............................................................................ filament, forms, notifications, support, tables  
  Version .............................................................................................................. v3.0.0-alpha67  
  Views ........................................................................................................... PUBLISHED: filament

**Reply by leandrocfe** [reply]
In your composer.json set:
`"minimum-stability": "dev"`

run

`composer require filament/upgrade:"^3.0@beta" --dev`
`vendor/bin/filament-v3`

follow the next steps..

**Reply by jcgmalave** [reply]
i get this:   Problem 1
    - Root composer.json requires filament/upgrade 3.0@beta, found filament/upgrade[v3.0.0-alpha56, ..., 3.x-dev] but it does not match the constraint.


Thanks a lot for the help :)

**Reply by leandrocfe** [reply]
run `composer update -W`
if you want send a DM on filament discord

**Reply by leandrocfe** [reply]
sorry Man, my bad. Actually you are running alpha

just run this

`composer require filament/filament:"^3.0@beta" -W`
`php artisan filament:install --panels`

**Reply by jcgmalave** [reply]
when i run: 

composer require filament/filament:"^3.0@beta" -W
php artisan filament:install --panels     i get the following error:

Problem 1
    - Root composer.json requires filament/filament 3.0@beta, found filament/filament[v0.1.0, v1.0.0, ..., 1.x-dev, v2.0.0-beta1, ..., 2.x-dev, v3.0.0-alpha1, ..., 3.x-dev] but it does not match the constraint.

**Reply by jcgmalave** [reply]
ive been able to do something with this:

composer require filament/upgrade:"v3.0.0-beta8" --dev

**Reply by leandrocfe** [reply]
Did it work?

**Reply by jcgmalave** [reply]
no, after that i run: vendor/bin/filament-v3 the process goes fine and then it says i have to run composer update, and while doing this it shows this: 

In AdminPanelProvider.php line 22:

  Class "Filament\PanelProvider" not found
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

**Reply by leandrocfe** [reply]
`composer require filament/filament:"^3.0@beta" -W`
It was supposed to work. Can we talk on Filament Discord? https://filamentphp.com/discord
Please send me a message, and after we will post the solution here.

**Reply by jcgmalave** [reply]
HI, how are you? i was able to install the latest beta version, but now im getting this related to taiwind:

i get the following error trying to upgrade v3:

[vite:css] C:\xampp\htdocs\modatex-chile-tres\resources\css\app.css:4:1: 'colors.gray.950 / 5%' does not exist in your theme config. 'colors.gray' has the following valid keys: '50', '100', '200', '300', '400', '500', '600', '700', '800', '900'

Any ideas?

thanks in advanced

**Reply by emjay64** [reply]
@jcgmalave I am facing the same issue:
```
In AdminPanelProvider.php line 22:
  Class "Filament\PanelProvider" not found 

``` 
did you get this resolved?

**Reply by danharrin** [reply]
What is your Filament version, check with `php artisan about`

**Reply by emjay64** [reply]
Version ........................................................................................................................... v3.0.0-alpha67

**Reply by danharrin** [reply]
```bash
composer require filament/filament:"^3.0-stable" -W
```

**Comment by JayrousJr** [comment]
I too get the same error

 In AdminPanelProvider.php line 22:

  Class "Filament\PanelProvider" not found

**Reply by danharrin** [reply]
You have the wrong version installed then

**Reply by Wan237** [reply]
Have you solved this problem?it cost me a day already!

**Reply by leandrocfe** [reply]
Are you updating it?

**Comment by Jigsaw5279** [comment]
I'm having the same issue on a fresh install of both Laravel 10 and Filament, copying commands from the docs.

![msrdc_CwBcQmciF3](https://github.com/filamentphp/filament/assets/3465075/06794957-d4a9-4abe-9b8b-b525eb20a6c4)

The classes just don't exist.

![msrdc_OT47T5tvsJ](https://github.com/filamentphp/filament/assets/3465075/a9c6cbf8-d834-46fe-976c-7d89d19f4470)

**Reply by danharrin** [reply]
Run `php artisan about`, what version are you on

**Reply by Jigsaw5279** [reply]
Environment ......................................................................................................................................  
  Application Name ......................................................................................................................... Laravel  
  Laravel Version .......................................................................................................................... 10.18.0  
  PHP Version ................................................................................................................................ 8.2.8  
  Composer Version ........................................................................................................................... 2.5.8  
  Environment ................................................................................................................................ local  
  Debug Mode ............................................................................................................................... ENABLED  
  URL ................................................................................................................................ crm.ddev.site  
  Maintenance Mode ............................................................................................................................. OFF  

  Cache ............................................................................................................................................  
  Config ................................................................................................................................ NOT CACHED  
  Events ................................................................................................................................ NOT CACHED  
  Routes ................................................................................................................................ NOT CACHED  
  Views ................................................................................................................................. NOT CACHED  

  Drivers ..........................................................................................................................................  
  Broadcasting ................................................................................................................................. log  
  Cache ....................................................................................................................................... file  
  Database ................................................................................................................................... mysql  
  Logs .............................................................................................................................. stack / single  
  Mail ........................................................................................................................................ smtp  
  Queue ....................................................................................................................................... sync  
  Session ..................................................................................................................................... file  

  Filament .........................................................................................................................................  
  Packages ................................................................................................... forms, notifications, support, tables  
  Version .................................................................................................................................. v3.0.17  
  Views .............................................................................................................................. NOT PUBLISHED

**Reply by sinnbeck** [reply]
@Jigsaw5279 What if you install this ?
```bash
composer require filament/filament:"^3.0-stable" -W
```
It seems the file is in this package, and it isn't required by the others. You are trying to use Panel which tells you to install that

https://filamentphp.com/docs/3.x/panels/installation

**Reply by Jigsaw5279** [reply]
That.. worked. I was absolutely sure I followed the docs since I copied the commands from there. I started installing all the other filament/* packages when I ran into that error. Somehow I must have ended up on the forms installation first?
Thanks anyways..

**Reply by Jigsaw5279** [reply]
Funny enough,` artisan about` still reports the same stats about Filament after running that command

**Reply by danharrin** [reply]
I've just seen the problem. You didn't actually install Filament panel builder. You just installed all the other packages. That class is part of the panel builder. You were on the correct version.

**Reply by Jigsaw5279** [reply]
Maybe adding a check if the PanelProvider exists in the install command would help?

**Reply by danharrin** [reply]
Good idea, please PR it :)

**Reply by Jigsaw5279** [reply]
Done #7673

**Comment by Nes-cmd** [comment]
I have also encountered this error. It happened after upgrading process was finished. Then I deleted the vendor folder then run `composer update` It worked.

**Comment by Wan237** [comment]
Problem solved, thanks for replying.

**Comment by Quantumben** [comment]
To fix this error run "composer require filament/filament:"^3.0-stable" --ignore-platform-reqs --with-all-dependencies"

**Comment by Erickhrs** [comment]
Guys, I resolved it by following these steps on a shared host (Hostinger):

1. Zip all my Laravel files (with Filament) and upload them to the host directory.
2. Update the database information in the `.env` file.
3. Add the `.htaccess` file:

```
RewriteEngine On

# Handle Authorization Header
RewriteCond %{HTTP:Authorization} .
RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

# Redirect Trailing Slashes If Not A Folder...
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_URI} (.+)/$
RewriteRule ^ %1 [L,R=301]

# Remove public URL from the path
RewriteCond %{REQUEST_URI} !^/public/
RewriteRule ^(.*)$ /public/$1 [L,QSA]

# Send Requests To Front Controller...
RewriteCond %{REQUEST_FILENAME} !-d
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule ^ index.php [L]
```

4. Access the host via SSH and run in the laravel directory:

```bash
composer --version
```

If the version is < 2.0, run the following commands:

```bash
php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
php -r "if (hash_file('sha384', 'composer-setup.php') === 'dac665fdc30fdd8ec78b38b9800061b4150413ff2e3b6f88543c636f7cd84f6db9189d43a81e5503cda447da73c7e5b6') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('composer-setup.php'); } echo PHP_EOL;"
php composer-setup.php
php -r "unlink('composer-setup.php');"
```

(This will create a `.phar` file)

5. Run the following commands:

```bash
php artisan key:generate
php artisan migrate
./composer.phar install
```

If it still doesn’t work, try:

```bash
./composer.phar dump-autoload
```

This will either resolve the issue or indicate what the problem is. 

In my case, the problem was:

```
[u369742331@br-asc-web1845 public_html]$ ./composer.phar dump-autoload
Generating optimized autoload files
Class App\Http\Controllers\Api\QuestionsController located in ./app/Http/Controllers/api/QuestionsController.php does not comply with PSR-4 autoloading standard (rule: App\ => ./app). Skipping.
Class App\Models\Answer located in ./app/Models/answer.php does not comply with PSR-4 autoloading standard (rule: App\ => ./app). Skipping.
Class App\Models\Question located in ./app/Models/question.php does not comply with PSR-4 autoloading standard (rule: App\ => ./app). Skipping.
Class App\Models\Comment located in ./app/Models/comment.php does not comply with PSR-4 autoloading standard (rule: App\ => ./app). Skipping.
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
```

So basically, what you need to do is fix all the file or folder names that don’t follow the PSR-4 standard. After fixing them, run `./composer.phar dump-autoload` again. It needs to run until all errors are resolved. 

Tip: If you don’t understand what is PSR-4, paste the error into ChatGPT. It will tell you which folders and files names need fixing.
